windows/table.cpp

```diff
@@ -0,0 +1,295 @@
+#include "uipriv_windows.hpp"
+
+//static void uiTableMinimumSize(uiWindowsControl *c, int *width, int *height);
+
+struct uiTable;
+
+struct uiTableModel {
+	uiTableModelHandler *mh;
+	std::vector<uiTable *> tables;
+};
+
+struct uiTableColumn {
+	uiTable *t;
+	WCHAR *name;
+	// don't really support parts (but this would part=>column mappings if we did)
+	int modelColumn;	// -1 = none
+};
+
+struct uiTable {
+	uiWindowsControl c;
+	uiTableModel *model;
+	HWND hwnd;
+	std::vector<uiTableColumn *> columns;
+};
+
+void *uiTableModelStrdup(const char *str)
+{
+	return strdup(str);
+}
+
+void *uiTableModelGiveColor(double r, double g, double b, double a)
+{
+	return 0;	// not implemented
+}
+
+uiTableModel *uiNewTableModel(uiTableModelHandler *mh)
+{
+	uiTableModel *m;
+
+	m = new uiTableModel();
+	m->mh = mh;
+	return m;
+}
+
+
+void uiFreeTableModel(uiTableModel *m)
+{
+	delete m;
+}
+
+void uiTableModelRowInserted(uiTableModel *m, int newIndex)
+{
+	LVITEMW item;
+
+	ZeroMemory(&item, sizeof (LVITEMW));
+	item.mask = 0;
+	item.iItem = newIndex;
+	item.iSubItem = 0; //?

```

Comment:
0 appears to be correct here. yes, since a subitem index of 0 means the item itself, not any of its subitems. (The leftmost column is not considered as "subitem" because of how listview is basically an extended listbox.)